microceph/ceph/crush.go

```diff
@@ -101,3 +102,46 @@ func setPoolCrushRule(pool string, rule string) error {
 	}
 	return nil
 }
+
+// setDefaultCrushRule sets the default crush rule for new pools
+func setDefaultCrushRule(rule string) error {
+	rid, err := getCrushRuleID(rule)
+	if err != nil {
+		return err
+	}
+	SetConfigItem(types.Config{
+		Key:   "osd_pool_default_crush_rule",
+		Value: rid,
+	})
+	return nil
+}
+
+// getDefaultCrushRule returns the current default crush rule for new pools
+func getDefaultCrushRule() (string, error) {
+	configs, err := GetConfigItem(types.Config{
+		Key: "osd_pool_default_crush_rule",
+	})
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(configs[0].Value), nil
+}
+
+// ensureCrushRules set up the crush rules for the automatic failure domain handling.
+func ensureCrushRules() error {
+	// Add a microceph default rule with failure domain OSD if it does not exist.
+	if !haveCrushRule("microceph_auto_osd") {
+		err := addCrushRule("microceph_auto_osd", "osd")
+		if err != nil {
+			return fmt.Errorf("Failed to add microceph default rule: %w", err)
```

Comment:
While this is a refactor, moving the method - I do think its worth s/rule/crush rule/ for avoidance of doubt. Also in the error message below.